src/Form/Type/RolesMatrixType.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\UserBundle\Form\Type;
+
+use Sonata\UserBundle\Security\RolesBuilder\ExpandableRolesBuilderInterface;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormTypeInterface;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Silas Joisten <silasjoisten@hotmail.de>
+ */
+final class RolesMatrixType extends AbstractType
+{
+    /**
+     * @var RolesBuilderInterface
+     */
+    private $rolesBuilder;
+
+    public function __construct(ExpandableRolesBuilderInterface $rolesBuilder)
```

Comment:
Here we have `ExpandableRolesBuilderInterface` and comment on variable points to `RolesBuilderInterface`